ompi/op/op.h

```diff
@@ -500,9 +514,11 @@ static inline bool ompi_op_is_valid(ompi_op_t * op, ompi_datatype_t * ddt,
  * optimization).  If you give it an intrinsic op with a datatype that
  * is not defined to have that operation, it is likely to seg fault.
  */
-static inline void ompi_op_reduce(ompi_op_t * op, const void *source,
-                                  void *target, size_t full_count,
-                                  ompi_datatype_t * dtype)
+static inline void ompi_op_reduce_stream(ompi_op_t * op, const void *source,
+                                         void *target, size_t full_count,
+                                         ompi_datatype_t * dtype,
+                                         int device,
```

Comment:
Yes, I wanted to avoid asking for the device over and over again. We talked about fat pointers that store that information. Once we have them we can change this back.